proposals/distribution.md

```diff
@@ -103,19 +103,17 @@ The following entries should be added to the [scope table][scope]:
 * “Specifying a distribution method”.
     This entry replaces part of the previous “Creating Reference spec for optional DNS based naming & distribution” and “Standardizing on a particular Distribution method” entries.
 
-    Retrieving image indexes covers the current “tag listing” (e.g. “what named manifests are in `library/busybox`?”), because tags are entries in the image format's [`manifests` array][manifests].
-    Other tag-listing endpoints needed for backwards-compatibility are therefore in scope as well.
+    Tag-listing (e.g. “what named manifests are in `library/busybox`?”) endpoints are in scope and required for backwards compatibility with clients.
+    External methods for listing tags remain compatible through fetching by content address, but are out of scope.
+    An endpoint for listing image repository names within a registry is considered out of scope for this specification.
 
-    Grouping image indexes in repositories is considered part of distribution policy or content management, which are out of scope for this entry's per-image action.
-    For example, “what images are under `library/`?” is out of scope for this project.
+    Managing the grouping of image repository names is considered part of distribution policy or content management, which are out of scope.
+    For example, “which image repositories are under `library/`?” is out of scope for this project.
 
     * What: Specifying a distribution method
     * In/Out/Future: In scope
     * Status: In progress (see opencontainers/distribution-spec)
-    * Description: Define a protocol for creating, retrieving, updating, and deleting objects defined in the [image specification][image-spec].
-        Listing repositories (like [`/v2/_catalog`][catalog]) is a multi-[image-index][] action, which is out of scope for this entry.
-        Managing groups of image indexes requires multi-[image-index][] actions, which are out of scope for this entry.
-        Listing image indexes within a group is a multi-[image-index][] action, which is out of scope for this entry.
+    * Description: Define a protocol for creating, retrieving, updating, and deleting content-addressable objects, such as those defined in the [image specification][image-spec].
```

Comment:
The distribution mandate needs to be broader than CAS.  I don't think this sentence is broad enough to justify including the mutable, name-agressable tag-listing and tag -> manifest endpoints, which you list as in-scope above.

Reply:
This sentence is describing the protocol which is based on content-addressable objects. Naming just maps to these objects, the objects themselves are not mutable. It is clearly listed up above that these mapping endpoints are in scope and will be supported. The protocol however can be used without this tagging.

Reply:
> This sentence is describing the protocol which is based on content-addressable objects.

The protocol uses content-addressable objects with a mutable, name-addressable layer on top (like Git's refs).  Of the endpoints [listed here][1], only `/v2/<name>/blobs/<digest>` is strictly content addressable.  The stuff under `/v2/<name>/blobs/uploads/` is how you feed data into CAS, so that's fine too.  But `/v2/<name>/manifests/<reference>` and `/v2/<name>/tags/list` are mutable and name-addressable, so you need some other wording to justify them.

> Naming just maps to these objects, the objects themselves are not mutable.

So add a sentence that includes this naming in the scope?  As long as any mentions of “naming” are clearly distinct from [the out-of-scope “… DNS based naming…” entry][2].

> It is clearly listed up above that these mapping endpoints are in scope and will be supported.

Right, but that text is just informative background that won't end up in the scope table.  I think we want normative language that will end up in the scope table to explain why tag listing and name-addressable references are in scope.

[1]: https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md#detail
[2]: https://github.com/opencontainers/tob/blame/e07b90ef134e4b1c4b852cc601f60e9107eecc47/proposals/distribution.md#L90-L101

Reply:
> So add a sentence that includes this naming in the scope?

Can you propose a sentence to add and we can discuss. I don't want to go back and forth on the correctness of the description. I am not sure how the scope table is generated, sounds like the previous sentences are just there for the proposal but unrelated to the scope table at https://www.opencontainers.org/about/oci-scope-table

Reply:
> Can you propose a sentence to add and we can discuss.

My previous attempt is what you're removing here ;).  I was trying to lean on image-spec indexes and `org.opencontainers.image.ref.name` to show why tag-listing and tag -> manifest-descriptor resolution were in scope.  It sounds like that's the basic idea, but @stevvooe wants phrasing that doesn't mention the image-spec primitives, and I don't have a rewording I like better yet.

Reply:
> I am not sure how the scope table is generated...

I think @caniszczyk just updates WordPress based on the instructions here after the TOB approves the proposal.

Reply:
> I was trying to lean on image-spec indexes and org.opencontainers.image.ref.name to show why tag-listing and tag -> manifest-descriptor resolution were in scope.

I get that, the problem is that image-spec indexes and that annotation have nothing to do with the distribution specification, so it was confusing. The annotation may be the same as what the registry has, but there is no relationship in the API. The reason the current tag endpoints are included are for backwards compatibility and there is discussions which need to be had about updating that interface and capabilities. However the point of this proposal is to take the specification as is and keep the scope narrowed down. The CAS part of the specification has already been widely agreed on and is good to have in the scope table, we can always broaden later as the discussion matures.

Reply:
> The reason the current tag endpoints are included are for backwards compatibility and there are discussions which need to be had about updating that interface and capabilities.

So add a sentence saying which endpoints are needed for backwards compat?  And making it clear that backwards-compat with those endpoints is in-scope while backwards-compat with `/v2/_catalog` is out-of-scope?

Reply:
This is why I was asking about the format of these entries, these sentences are already there. Are you saying we should just move the sentences from before the `What` to under `Description`? The previous version also had in/out of scope language in two places, it is really not clear what this section is trying to accomplish with the formatting.

Reply:
> This is why I was asking about the format of these entries, these sentences are already there. Are you saying we should just move the sentences from before the `What` to under `Description`? The previous version also had in/out of scope language in two places, it is really not clear what this section is trying to accomplish with the formatting.

My intention with the formatting was:

```markdown
* “{What entry}”.
    {Detailed informative discussion, intended for the proposal only}.

    * What: {What entry}
    * In/Out/Future: {Scope entry}
    * Status: {Status entry}
    * Description: {Normative definiton}
    * Why: {High-level informative motivation}
```

If you want to drop the detailed-informative-discussion section entirely because you find it not-very-informative, that's fine with me.  For the “Description” entry, how about something like:

```markdown
    * Description: Define a protocol for creating, retrieving, updating, and deleting content-addressable objects, such as those defined in the [image specification][image-spec].
        Also define, as an optional layer, the `/v2/<name>/tags/list` and `/v2/<name>/manifests/<reference>` endpoints needed for backwards compatibility with existing clients.
```